pkg/modules/solution/solveintents.go

```diff
@@ -0,0 +1,158 @@
+// Package solution sends the received bundler batch of Intent UserOperations
+// to the Solver to solve the Intent and fill-in the EVM instructions.
+//
+// This implementation makes 1 attempt for each Intent userOp to be solved.
+//
+// Solved userOps update the received bundle
+// All other returned statuses result in dropping those userOps
+// from the batch.
+// Received are treated as expired because they may have been compressed to
+// Solved Intents.
+//
+// The Solver may return a subset and in different sequence the UserOperations
+// and a matching occurs by the hash value of each UserOperation to the bundle
+// UserOperation.
+package solution
+
+import (
+	"bytes"
+	"math/big"
+	"net/http"
+	"time"
+	"unsafe"
+
+	"github.com/blndgs/model"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/goccy/go-json"
+	"github.com/pkg/errors"
+
+	"github.com/stackup-wallet/stackup-bundler/pkg/modules"
+	"github.com/stackup-wallet/stackup-bundler/pkg/userop"
+)
+
+// userOpHashID is the hash value of the UserOperation
+type opHashID string
+
+// batchOpIndex is the index of the UserOperation in the Bundler batch
+type batchOpIndex int
+
+// batchIntentIndices buffers the mapping of the UserOperation hash value -> the index of the UserOperation in the batch
+type batchIntentIndices map[opHashID]batchOpIndex
+
+type EntryPointIntents struct {
```

Comment:
we should be very careful with terminology
EntryPoint is taken already, and this is confusing

Reply:
The Bundler has the ability to support multiple Entrypoint contracts and creates batches per Entrypoint. Currently, it only initializes with version 0.6. The name was chosen because it reflects this relationship batch-> 1entrypoint, and the fact that you noticed it is the reason behind the name. Let me know if there is anything I can do to make it less confusing for you.